app/channels/room_channel.rb

```diff
@@ -13,16 +15,32 @@ def send_message(data)
     chat = get_chat(room_id)
     raise 'No message!' if message.blank?
 
-    Message.create!(chat: chat, user: sender, text: message)
+    Message.create!(chat: chat, user: sender, text: message, seen: other_user_connected?)
     payload = {
-      room_id: room_id, content: message, sender: sender, participants: convo.users
+      room_id: room_id, content: message, sender: sender, participants: chat.users
     }
     ActionCable.server.broadcast("ChatRoom-#{room_id}", payload)
   end
 
+  def unsubscribed
+    is_a = (current_user == chat.user_a)
+    is_a ? chat.update(user_a_connected: false) : chat.update(user_b_connected: false)
+  end
+
   private
 
   def get_chat(room_code)
     Chat.find(room_code)
```

Comment:
I don't know what the rest of the guys think, but maybe is worth to make use of advantages of memoization here, like:
```
attr_reader :chat

def initialize(room_code)
    @room_code = room_code
end
...
def chat
    @chat ||= Chat.find(@room_code)
end
```

Reply:
Thank you, that dried my code.